app/src/main/java/com/tomtom/sdk/examples/navigation/NavigationExamplesActivity.kt

```diff
@@ -11,15 +11,121 @@
+    private lateinit var routePlanningOptions: RoutePlanningOptions
+    private lateinit var route: Route
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
         binding = ActivityNavigationExamplesBinding.inflate(layoutInflater)
         setContentView(binding.root)
+        planRoute()
+    }
+
+    private fun planRoute() {
+        routePlanner = OnlineRoutePlanner.create(context = this, apiKey = TOMTOM_API_KEY)
+        val amsterdam = GeoPoint(latitude = 52.37616, longitude = 4.90828)
+        val rotterdam = GeoPoint(latitude = 51.90546, longitude = 4.46662)
+        val itinerary = Itinerary(origin = amsterdam, destination = rotterdam)
+        routePlanningOptions = RoutePlanningOptions(
+            itinerary = itinerary,
+            guidanceOptions = GuidanceOptions(progressPoints = ProgressPoints.All),
+            vehicle = Vehicle.Car()
+        )
+        routePlanner.planRoute(
+            routePlanningOptions,
+            object : RoutePlanningCallback {
+                override fun onSuccess(result: RoutePlanningResponse) {
+                    route = result.routes.first()
+                    startNavigation()
+                }
+
+                override fun onFailure(failure: RoutingFailure) {
+                    Log.e(TAG, "Unable to calculate a route: " + failure.message)
+                }
+
+                override fun onRoutePlanned(route: Route) = Unit
+            }
+        )
+    }
+
+    private fun createLocationProvider(): LocationProvider {
+        val routeGeoLocations = route.geometry.map { GeoLocation(it) }
+        val simulationStrategy = InterpolationStrategy(routeGeoLocations)
+        return SimulationLocationProvider.create(strategy = simulationStrategy)
+            .also { locationProvider ->
+                locationProvider.enable()
+            }
+    }
+
+    private fun createNavigationConfiguration(): Configuration {
+        val navigationTileStore = NavigationTileStore.create(
+            context = this,
+            navigationTileStoreConfig = NavigationTileStoreConfiguration(apiKey = TOMTOM_API_KEY)
+        )
+        val routeReplanner = OnlineRouteReplannerFactory.create(routePlanner)
+        val locationProvider = createLocationProvider()
+        return Configuration(
+            context = this,
+            locationProvider = locationProvider,
+            navigationTileStore = navigationTileStore,
+            routeReplanner = routeReplanner
+        )
+    }
+
+    private fun startNavigation() {
+        val configuration = createNavigationConfiguration()
+        val tomTomNavigation = OnlineTomTomNavigationFactory.create(configuration)
+
+        tomTomNavigation.addProgressUpdatedListener(progressUpdatedListener)
+
+        val routePlan = RoutePlan(route, routePlanningOptions)
+        val navigationOptions = NavigationOptions(routePlan)
+        tomTomNavigation.start(navigationOptions)
```

Comment:
We should stop and close navigation when this activity is destroyed

Reply:
Done.